Changes towards GWT compatibility

uruuru opened the pull request:
Not yet ready to merge - I fixed a number of issues. See below for more. 

@le-cds feel free to fix the issues and change the current fixes if you see a better way. 

Also, I added a `TODO` where a statement seemed to have no effect. Can you please double check that?

```
      [ERROR] Errors in 'file:/workspace/elkjs/build/src/elk/org/eclipse/elk/core/util/ElkUtil.java'
         [ERROR] Line 1013: No source code is available for type java.util.regex.Pattern; did you forget to inherit a required module?

      [ERROR] Errors in 'file:/workspace/elkjs/build/src/elk/org/eclipse/elk/core/util/BasicProgressMonitor.java'
         [ERROR] Line 477: The method format(String, long, char, char, String) is undefined for the type String
         [ERROR] Line 499: The method format(String, int, String) is undefined for the type String
```

le-cds commented on plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/LoggedGraph.java:
```diff
@@ -108,10 +112,14 @@ public String getTag() {
     public String serialize() {
+	// TODO what is the effect?
+	// elkjs-exclude-start
         graphType.isTypeCompatible(graph);
```
There's no effect here. I think I expected the method to throw an exception. I changed the implementation to throw an exception when constructing a new LoggedGraph instance.

uruuru commented:
With the latest changes elkjs builds with the current master, and the tests run succesfully. 

I'll have to remove these whitespace changes automatically inserted by gitpod though ~.~.

uruuru commented:
I cleaned the diffs and slightly adjusted the fixes, which should be fine now. 

I also added a workaround to allow execution time measurements in js (3c82e4be62a7289b5891d4933d00a2f6ae1c36b3). @le-cds please let me know if you're fine with this or have a better idea. 

Any other comments? Otherwise I'd like to merge it.

le-cds commented:
> I also added a workaround to allow execution time measurements in js (3c82e4b). @le-cds please let me know if you're fine with this or have a better idea.

No, I'm fine with this. :)

> Otherwise I'd like to merge it.

Please go ahead.
